src/main/java/com/lb/application/TripCoordinator.java

```diff
@@ -0,0 +1,135 @@
+package com.lb.application;
+
+import akka.javasdk.client.ComponentClient;
+import com.lb.ai.models.TripAgentChatModel;
+import com.lb.ai.tools.*;
+import com.lb.api.*;
+import java.io.ByteArrayInputStream;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ai.chat.client.ChatClient;
+
+public class TripCoordinator {
+
+  private static final Logger log = LoggerFactory.getLogger(TripCoordinator.class);
+
+  private final ComponentClient componentClient;
+  private final ChatClient chatClient;
+
+  public TripCoordinator(TripAgentChatModel tripAgentChatModel, ComponentClient componentClient) {
+    this.componentClient = componentClient;
+    this.chatClient = ChatClient.create(tripAgentChatModel);
+  }
+
+  public CompletionStage<String> requestTrip(String question) {
+    // TODO use virtual threads
+    // TODO Check out why when using CompletableFuture Jackson gets hijacked and parsing fails.
+    // TODO search for already existing flights before using the "external" API.
+    log.info("looking for flights");
+
+    String responseFlights =
+        chatClient
+            .prompt(
+                String.format(
+                    """
+                       find ONLY flights with the following constraints %s. Ignore any constraints that don't refer flights
+                       Parse the flights as a JSON such they fit a schema parseable to a Java class like this:
+                       FlightAPIResponse(String id, String from, String to, ZonedDateTime departure, ZonedDateTime arrival, int price)
+                       Create the JSON such it has only a list of FlightAPIResponse, do not add any other field
```

Comment:
I guess it understands how to structure that json? Otherwise it seems good to include an example json in the request, see https://github.com/akka-samples/travel-agent/pull/1/files#diff-8593cb8a9ee4e3d50ddbe03385f74fa42f7e3699e64557ce8ce71e493fd6b244R47